starport-core/src/main/scala/com/krux/starport/metric/MetricSettings.scala

```diff
@@ -3,38 +3,107 @@ package com.krux.starport.metric
 import com.typesafe.config.Config
 
+/**
+ * A service trait that is used by private case class constructors.
+ */
+trait MetricSettingsImpl extends Serializable {
 
-class MetricSettings(val config: Config) extends Serializable {
+  def getReporter(registry: MetricRegistry): ScheduledReporter
 
-  val hosts = config.getStringList("hosts")
+}
 
-  val port = config.getInt("port")
+/**
+ * A sealed trait for metric engine settings. We want this to be sealed as there should only be a finite number of
+ * reporting engine options available.
+ */
+sealed trait MetricSettings {
 
-  val metricPrefix = config.getString("prefix")
+  /** Set the default duration to be in seconds */
+  final val DefaultDuration = TimeUnit.SECONDS
 
-  def getGraphite: Graphite =
-    new Graphite(hosts.get(Random.nextInt(hosts.size)), port)
+  def getReporter(config: Config, registry: MetricRegistry): ScheduledReporter = this match {
+      case GraphiteReporterSettings => GraphiteReporterSettings(config).getReporter(registry)
+      case CloudWatchReporterSettings => CloudWatchReporterSettings(config).getReporter(registry)
+  }
 
-  def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter.forRegistry(registry)
-    .prefixedWith(metricPrefix)
-    .convertDurationsTo(MetricSettings.DefaultDuration)
-    .build(getGraphite)
+  def getDefaultReporter(registry: MetricRegistry): ScheduledReporter = ConsoleReporter
+    .forRegistry(registry)
+    .convertDurationsTo(DefaultDuration)
+    .build()
 
 }
 
-object MetricSettings {
+final case object GraphiteReporterSettings extends MetricSettings {
 
-  final val DefaultDuration = TimeUnit.SECONDS
+  def apply(config: Config): MetricSettingsImpl = GraphiteReporterSettingsImpl(config)
+
+  private case class GraphiteReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val hosts = config.getStringList("hosts")
+
+    val port = config.getInt("port")
+
+    val metricPrefix = config.getString("prefix")
+
+    def getGraphite: Graphite =
+      new Graphite(hosts.get(Random.nextInt(hosts.size)), port)
+
+    def getReporter(registry: MetricRegistry): ScheduledReporter = GraphiteReporter
+      .forRegistry(registry)
+      .prefixedWith(metricPrefix)
+      .convertDurationsTo(DefaultDuration)
+      .build(getGraphite)
+
+  }
 
-  def getReporter(configOpt: Option[Config], registry: MetricRegistry): ScheduledReporter =
-    configOpt.map(new MetricSettings(_).getReporter(registry))
-      .getOrElse(
-        ConsoleReporter.forRegistry(registry).convertDurationsTo(DefaultDuration).build()
-      )
 }
+
+final case object CloudWatchReporterSettings extends MetricSettings {
+
+  def apply(config: Config): MetricSettingsImpl = CloudWatchReporterSettingsImpl(config)
+
+  private case class CloudWatchReporterSettingsImpl(config: Config) extends MetricSettingsImpl {
+
+    val awsRegion: Region = config.getString("region").asInstanceOf[Region]
```

Comment:
I believe this will throw an exception on runtime because String can't be cast to Region. We should use `Region.of(...)` instead.

Reply:
Got it. Nice catch.